homeassistant/components/update/device_action.py

```diff
@@ -0,0 +1,107 @@
+"""Provides device actions for update entities."""
+from __future__ import annotations
+
+from typing import Final
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_ENTITY_ID,
+    CONF_TYPE,
+)
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import entity_registry
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import get_supported_features
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    ATTR_BACKUP,
+    DOMAIN,
+    SERVICE_INSTALL,
+    SERVICE_SKIP,
+    UpdateEntityFeature,
+)
+
+ACTION_TYPES: Final[set[str]] = {"install", "skip"}
+
+ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_TYPE): vol.In(ACTION_TYPES),
+        vol.Required(CONF_ENTITY_ID): cv.entity_domain(DOMAIN),
+        vol.Optional("backup"): cv.boolean,
+    }
+)
+
+
+async def async_get_actions(
+    hass: HomeAssistant, device_id: str
+) -> list[dict[str, str]]:
+    """List device actions for update devices."""
+    registry = await entity_registry.async_get_registry(hass)
+    actions = []
+
+    # Get all the integrations entities for this device
+    for entry in entity_registry.async_entries_for_device(registry, device_id):
+        if entry.domain != DOMAIN:
+            continue
+
+        supported_features = get_supported_features(hass, entry.entity_id)
+
+        base_action = {
+            CONF_DEVICE_ID: device_id,
+            CONF_DOMAIN: DOMAIN,
+            CONF_ENTITY_ID: entry.entity_id,
```

Comment:
Since this is a new action, should we just use the UUID instead?

Reply:
Yeah, I think that we should use uuids everywhere.

Reply:
🤷